vision/standard_object/odlc_text_detection.py

```diff
@@ -0,0 +1,155 @@
+    final_img: Image = cv2.dilate(eroded_img, kernel=kernel, iterations=1)
+
+    return final_img
+
+
+def crop_image(uncropped_img: Image, bounds: BoundingBox) -> Image:
+    """
+    Crops the passed image according to the passed bounding box.
+
+    Parameters
+    ----------
+    uncropped_img: Image
+        The image to be cropped.
+    bounds: BoundingBox
+        The bounding box that has the cords to crop around.
+
+    Returns
+    -------
+    cropped_img: Image
+        The image after it has been cropped.
+    """
+
+    x_min: int
+    x_max: int
+    x_min, x_max = bounds.get_x_extremes()
+
+    y_min: int
+    y_max: int
+    y_min, y_max = bounds.get_y_extremes()
+
+    crop_img: Image = uncropped_img[y_min:y_max, x_min:x_max, :]
+
+    return crop_img
+
+
+def text_detection(base_img: Image) -> str:
+    """
+    Detects text on the passed image.
+
+    Parameters
+    ----------
+    base_img: Image
+        The image that you want to detect the text on.
+
+    Returns
+    -------
+    found_text: str
+        A single character string with the character detected in the image.
+    """
+
+    found_text: str  # Text to be returned, one character
+
+    found_text_info: dict[str, list[Any]] = pytesseract.image_to_data(
+        base_img, output_type=pytesseract.Output.DICT, config="--psm 10"
+    )
+
+    text: str
+    character: str
+
+    for text in found_text_info["text"]:
+        for character in text:
+            if (
+                character.isalnum() and character.isupper()
+            ):  # Text needs to be upercase and alphanumeric
+                found_text = character
+
+    return found_text
+
+
+def get_odlc_text(starting_image: Image, text_bounds: BoundingBox) -> str:
+    """
+    Detects text in the passed image at the location specified by the passed bounding box.
+
+    Parameters
+    ----------
+    starting_image: Image
+        The image to detect the text on.
+    text_bounds: BoundingBox
+        The bounding box specifying were in the image to look for text.
+
+    Returns
+    -------
+    detected_text: str
+        A string containing the character that was detected.
+    """
+
+    cropped_img: Image = crop_image(starting_image, text_bounds)
+
+    preprocessed_img: Image = text_detection_pre_processing(cropped_img)
+
+    detected_text: str = text_detection(preprocessed_img)
+
+    return detected_text
+
+
+if __name__ == "__main__":
+    import sys
+
+    img: npt.NDArray[np.uint8] = cv2.imread(
+        sys.argv[1]
+    )  # Run with something like 'python3 odlc_text_detection.py image.jpg'
+
+    # Testing coordinates
+    test_bounds: BoundingBox = BoundingBox(
+        ((740, 440), (820, 440), (820, 500), (740, 500)), ObjectType.STD_OBJECT
+    )  # Coords for the A in the star of the 2022 image
+
+    read_text: str = get_odlc_text(img, test_bounds)
+
+    print("The following character was detected:")
+
+    for i in enumerate(read_text):
+        print(i[1])
```

Comment:
`read_text` is a string of the detected character, so you wouldn't need this loop.

Also, you are using enumerate kind of weird. Typically, you would do `for index, data in enemerate(iterable):`, but you instead assigning i to a tuple. Doesn't really matter since you don't need the loop in the first place, but I thought I'd point it out.